e2e/tests/dnt.spec.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+const { test, expect } = require("@playwright/test");
+const config = require("../config");
+const {
+  generateUserCredentials
+} = require("../scripts/utils.js");
+
+const REGISTERED_USER_CREDENTIALS = generateUserCredentials();
+
+const registerUser = async (page) => {
+  await page.goto(config.RETAIL_APP_HOME + "/registration");
+
+  const registrationFormHeading = page.getByText(/Let's get started!/i);
+  await registrationFormHeading.waitFor();
+
+  await page
+    .locator("input#firstName")
+    .fill(REGISTERED_USER_CREDENTIALS.firstName);
+  await page
+    .locator("input#lastName")
+    .fill(REGISTERED_USER_CREDENTIALS.lastName);
+  await page.locator("input#email").fill(REGISTERED_USER_CREDENTIALS.email);
+  await page
+    .locator("input#password")
+    .fill(REGISTERED_USER_CREDENTIALS.password);
+
+  await page.getByRole("button", { name: /Create Account/i }).click();
+
+  await expect(
+    page.getByRole("heading", { name: /Account Details/i })
+  ).toBeVisible();
+
+  await expect(
+    page.getByRole("heading", { name: /My Account/i })
+  ).toBeVisible();
+}
+test.describe("Consent Tracking form works as expected", () => {
+  test.beforeEach(async ({ page }) => {
+    await page.context().clearCookies();
+    await page.goto(config.RETAIL_APP_HOME);
+  });
+
+  test("Closing form when defaultDnt exists sets DNT to defaultDnt", async ({ page }) => {
+    const modalSelector = '[aria-label="Close consent tracking form"]'
+    page.locator(modalSelector).waitFor()
+    await expect(page.getByText(/Tracking Consent/i)).toBeVisible({timeout: 10000});
+    await page.locator('button[aria-label="Close consent tracking form"]').click();
+    
+    const cookies = await page.context().cookies();
```

Comment:
There are automated tests ([here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/de48df8054aaa774eb51d92688711f5260bac8e2/packages/commerce-sdk-react/src/auth/index.test.ts#L436) and [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/de48df8054aaa774eb51d92688711f5260bac8e2/packages/commerce-sdk-react/src/auth/index.test.ts#L571)) to test the expected behavior of setting the DNT cookie when the value of defaultDnt is different or doesn't exist. It doesn't seem like there is a straightforward way to mock constants (in the constants file) with playwright, but this tests the functionality of the use of defaultDnt and its application with the form end to end.

Reply:
Thanks I simplified and consolidated the tests into one flow, from the initial interaction of the form to login and logout.